### PR TITLE
allow-custom-options-in-order-by

### PR DESCRIPTION
**Pull request by hiagolcm**
#### What problem is this solving?

The sort options in `OrderBy` are hard coded. This PR aims to offer the `customOptions` prop to allow custom options.

It is also related to https://github.com/vtex-apps/store-discussion/issues/101

#### How should this be manually tested?

[workspace](https://customorderbyoptions--biggy.myvtex.com/)

#### Screenshots or example usage
In this example only three options are available.

```
"order-by.v2#custom": {
    "props": {
      "customOptions": [
        {
          "value": "",
          "label": "store/ordenation.release.date"
        },
        {
          "value": "OrderByTopSaleDESC",
          "label": "store/ordenation.sales"
        },
        {
          "value": "OrderByReleaseDateDESC",
          "label": "store/ordenation.release.date"
        }
      ]
    }
  }
```
image

#### Types of changes

* [ ] Bug fix (a non-breaking change which fixes an issue)
* [x] New feature (a non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

**Comment by victorhmp** [comment]
Hey @hiagolcm!

So glad to see this PR since I was about to start implementing this! I have a few suggestions regarding the API for this feature that I think would make it a bit more user-friendly (and maybe even reduce the change of something going wrong).

I would not expect the user to write out the options that they want to show exactly as it is in the code, as this is a bit error-prone. Since we don't support any custom `orderBy` value, I would expect the user to write an array of values that they want to enable, and that's it. No need to pass in the `label` for example and the values you would expect to be inside the `customOptions` array are an `enum`, right? Much more predictable! It would go something like:

```json
"order-by.v2#custom": {
    "props": {
      "customOptions": [ 
          "OrderByReleaseDateDESC", "OrderByTopSaleDESC", "OrderByReleaseDateDESC"
      ]
    }
  }
```

**Comment by hiagolcm** [comment]
> Hey @hiagolcm!
> 
> So glad to see this PR since I was about to start implementing this! I have a few suggestions regarding the API for this feature that I think would make it a bit more user-friendly (and maybe even reduce the change of something going wrong).
> 
> I would not expect the user to write out the options that they want to show exactly as it is in the code, as this is a bit error-prone. Since we don't support any custom `orderBy` value, I would expect the user to write an array of values that they want to enable, and that's it. No need to pass in the `label` for example and the values you would expect to be inside the `customOptions` array are an `enum`, right? Much more predictable! It would go something like:
> 
> ```json
> "order-by.v2#custom": {
>     "props": {
>       "customOptions": [ 
>           "OrderByReleaseDateDESC", "OrderByTopSaleDESC", "OrderByReleaseDateDESC"
>       ]
>     }
>   }
> ```

@victorhmp

I'm glad to be helpful and thank you for your suggestions.

The main reason of this PR is to provide a way to integrate new search engines. In the current situation, VTEX-IO components only supports the default VTEX search.
Here is another PR I submitted with the same goal https://github.com/vtex-apps/store-components/pull/589

This PR aims to make the sort options customizable. By customizable I mean: provide new options (from a new search engine) or hide VTEX default options. In this way, if only the value is provided in the props, as you suggested, new search engines will be limited to the VTEX options.

Although your suggestion doesn't cover all the cases I initially proposed, I will keep it. Configurations with so many details of the implementation should not be written in the blocks.

In a conversation with the store-components team, we decided to reduce the scope of this PR and only provide a way to hide VTEX options (not create new ones anymore). Thus, I changed the prop from `customOptions` to `hiddenOptions`.

File: react/OrderByFlexible.js [hunk 1]
```diff
@@ -5,15 +5,15 @@ import OrderBy from './OrderBy'
...
+const withSearchPageContextProps = Component => (props) => {
   const { orderBy } = useSearchPage()
   if (orderBy == null) {
     return null
   }
 
   return (
     <div className={styles['orderBy--layout']}>
-      <Component orderBy={orderBy} />
+      <Component {...props} orderBy={orderBy} />
```
Comment (lbebber) [inline react/OrderByFlexible.js]: what is the purpose of this change?

Comment (lbebber) [reply react/OrderByFlexible.js]: by that I mean, can you just pass down the specific prop you need?

Comment (hiagolcm) [reply react/OrderByFlexible.js]: You mean this?

```
const withSearchPageContextProps = Component => ({hiddenOptions}) => {
 // omitted
<Component hiddenOptions={hiddenOptions} orderBy={orderBy} />
// omitted
}
```

File: README.md [hunk 2]
```diff
@@ -397,6 +397,25 @@ Notice that the default behavior for your store will be the one defined by the `
 
 Also, you can configure the product summary that is defined on search-result. See [here](https://github.com/vtex-apps/product-summary/blob/master/README.md#configuration) the Product Summary API.
 
+##### `order-by` block
```
Comment (iaronaraujo) [inline README.md]: I think you can remove this file since we moved the readme to the docs folder
